ur_robot_driver/src/dashboard_client_ros.cpp

```diff
@@ -241,111 +343,129 @@ bool DashboardClientROS::handleSavedQuery(ur_dashboard_msgs::srv::IsProgramSaved
 {
-  resp->answer = this->client_.sendAndReceive("safetymode\n");
-  std::smatch match;
-  std::regex expected("Safetymode: (.+)");
-  resp->success = std::regex_match(resp->answer, match, expected);
-  if (resp->success)
+  try
   {
-    if (match[1] == "NORMAL")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::NORMAL;
-    }
-    else if (match[1] == "REDUCED")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::REDUCED;
-    }
-    else if (match[1] == "PROTECTIVE_STOP")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::PROTECTIVE_STOP;
-    }
-    else if (match[1] == "RECOVERY")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::RECOVERY;
-    }
-    else if (match[1] == "SAFEGUARD_STOP")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::SAFEGUARD_STOP;
-    }
-    else if (match[1] == "SYSTEM_EMERGENCY_STOP")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::SYSTEM_EMERGENCY_STOP;
-    }
-    else if (match[1] == "ROBOT_EMERGENCY_STOP")
-    {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::ROBOT_EMERGENCY_STOP;
-    }
-    else if (match[1] == "VIOLATION")
+    resp->answer = this->client_.sendAndReceive("safetymode\n");
+    std::smatch match;
+    std::regex expected("Safetymode: (.+)");
+    resp->success = std::regex_match(resp->answer, match, expected);
+    if (resp->success)
     {
-      resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::VIOLATION;
+      if (match[1] == "NORMAL")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::NORMAL;
+      }
+      else if (match[1] == "REDUCED")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::REDUCED;
+      }
+      else if (match[1] == "PROTECTIVE_STOP")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::PROTECTIVE_STOP;
+      }
+      else if (match[1] == "RECOVERY")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::RECOVERY;
+      }
+      else if (match[1] == "SAFEGUARD_STOP")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::SAFEGUARD_STOP;
+      }
+      else if (match[1] == "SYSTEM_EMERGENCY_STOP")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::SYSTEM_EMERGENCY_STOP;
+      }
+      else if (match[1] == "ROBOT_EMERGENCY_STOP")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::ROBOT_EMERGENCY_STOP;
+      }
+      else if (match[1] == "VIOLATION")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::VIOLATION;
+      }
+      else if (match[1] == "FAULT")
+      {
+        resp->safety_mode.mode = ur_dashboard_msgs::msg::SafetyMode::FAULT;
+      }
+      // The following are only available in SafetyStatus from 5.5 on
```

Comment:
Should this be parameterized by the firmware version? Would this be feasible in the future. If so, could you open a follow-up issue.